From fc394edb9b94e1445a7e5f6f3d9a252634e9fc17 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 09 Mar 2022 08:51:26 +0200
Subject: [PATCH] Revert "[SW-58071] gaudi3: add scratchpad init routine"

Its been decided that scratchpad PSOC memory will be initialized
by HW upon cold reset. Due to this change, ROM doesn't need to
perform scratchpad initialization upon cold reset.

This reverts commit 8e2998f57387d317607ffbf78ac68171600516ef.

Change-Id: I0ce5e63d336ca1e195aa3ca531366a639ad7c69a
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 80c9ba5..80d48e4 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -83,7 +83,7 @@
 	regmap_writel(reg, mmCOLD_RST_FLOPS_2);
 }
 
-static void arcboot_rom_init_sp_sram(void)
+void arcboot_rom_init_sp_sram(void)
 {
 	uint32_t mem_cfg, cpy_sts;
 	uint8_t retry = HL_INIT_SP_SRAM_RETRIES;
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index b502b3b..b32a156 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -5,7 +5,6 @@
  */
 
 #include <arcboot.h>
-#include <delay.h>
 #include <gaudi3_fw_if.h>
 #include <efuse.h>
 #include <dev_spi_flash.h>
@@ -109,43 +108,6 @@
 	arcboot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
 }
 
-static void arcboot_rom_init_sp_sram(void)
-{
-	struct gaudi3_cold_rst_data cold_rst_data;
-	uint32_t init_sts;
-	uint8_t retry = HL_INIT_SP_SRAM_RETRIES;
-
-	/* Perform scratchpad init only on cold reset */
-	cold_rst_data.data = regmap_readl(mmCOLD_RST_FLOPS_2);
-	if (cold_rst_data.spsram_init_done)
-		return;
-
-	regmap_writel(D0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_START_M,
-			mmD0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL);
-	while (1) {
-		init_sts = regmap_readl(mmD0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL);
-		if (init_sts & D0_PSOC_GLOBAL_CONF_SCRATCHPAD_INIT_CTRL_DONE_M) {
-			cold_rst_data.spsram_init_done = 1;
-			regmap_writel(cold_rst_data.data, mmCOLD_RST_FLOPS_2);
-			break;
-		}
-
-		/*
-		 * Do not wait indefinitely on copy status
-		 * Wait for 1 milliseconds (100 retries with 10 micro-
-		 * seconds sleep), else bailout.
-		 */
-		retry--;
-		if (!retry) {
-			arcboot_rom_status_update(
-					PROGRESS_INIT_SP_SRAM_NOT_COMPLETED);
-			break;
-		}
-
-		udelay(HL_INIT_SP_SRAM_UDELAY);
-	}
-}
-
 void arcboot_rom_init(void)
 {
 	arcboot_rom_global.die_id = regmap_readl(mmD0_PARC_LOCAL_CFG_DIE_ID);
@@ -158,9 +120,6 @@
 	arcboot_rom_prstn_init();
 
 	arcboot_rom_read_cold_flops();
-
-	/* TODO: check whether required for die1 */
-	arcboot_rom_init_sp_sram();
 
 	spi_flash_init();
 
